packages/flutter/lib/src/widgets/animated_cross_fade.dart

```diff
@@ -24,6 +24,19 @@ enum CrossFadeState {
   showSecond,
 }
 
+/// Signature for the [AnimatedCrossFade.layoutBuilder] callback.
+///
+/// The `topChild` is the child fading in, which is normally drawn on top. The
+/// `bottomChild` is the child fading out, normally drawn on the bottom.
+///
+/// For good performance, the returned widget tree should contain both the
+/// `topChild` and the `bottomChild`; the depth of the tree, and the types of
+/// the widgets in the tree, from the returned widget to each of the children
+/// should be the same; and where there is a widget with multiple children, the
+/// top child and the bottom child should be keyed using the provided
+/// `topChildKey` and `bottomChildKey` keys respectively.
```

Comment:
We want to avoid using global keys if at all possible. They're expensive.

You can always convert a LocalKey to a GlobalKey if you really need a global key on the receiving end.